### PR TITLE
Refactor accelerator profile handling

### PR DESCRIPTION
**Pull request by Gkrumbach07**
https://issues.redhat.com/browse/RHOAIENG-13106

## Description

This pull request refactors the handling of accelerator profiles and introduces a new custom hook useAcceleratorProfileForm to improve code organization and reusability.

## How Has This Been Tested?

1. try adding, editing, removing an accelerator profile from workbenches, notebook server, deploy nims modal, serving runtime modal, kserve modal

## Test Impact

none, this is a internal code clarity change

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by jpuzz0** [comment]
Testing out the accelerator profile dropdown for workbenches, after selecting an accelerator that isn't "None", then creating that workbench, and opening the Edit page for that created workbench afterwards, the previous selected accelerator doesn't seem selected any longer. I checked the main branch and this doesn't seem to be a problem there.

**Comment by Gkrumbach07** [comment]
@jpuzz0 I found the fix, it was not setting the select field default due to it be async to get the default

**Comment by pnaik1** [comment]
for accelerator profile  the workbench I see a new option `existing settings` when I click on none and create the workbench and edit the created workbench I see `existing settings` selected instead of `none`

File: frontend/src/utilities/useAcceleratorProfileFormState.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+};
+
+export type UseAcceleratorProfileFormResult = {
+  initialState: AcceleratorProfileState;
+  formData: AcceleratorProfileFormData;
+  setFormData: UpdateObjectAtPropAndValue<AcceleratorProfileFormData>;
+  resetFormData: () => void;
+  loaded: boolean;
+  loadError: Error | undefined;
+  refresh: () => Promise<AcceleratorProfileState | undefined>;
```
Comment (DaoDaoNoCode) [inline frontend/src/utilities/useAcceleratorProfileFormState.ts:21]: These 3 props are never used in the code, is there any need to return them?

**Review by DaoDaoNoCode — APPROVED** [review]
Tested and didn't find any issues so far. Left one small comment but that's not very important.
/lgtm

File: frontend/src/utilities/useAcceleratorProfileFormState.ts (hunk 1 above)
Comment (Gkrumbach07) [reply frontend/src/utilities/useAcceleratorProfileFormState.ts:21]: just to expose it if its needed elsewhere. we may use it in the future for another feature.

**Review by DaoDaoNoCode — APPROVED** [review]
/lgtm

**Pull request merged**